### PR TITLE
Change solvency ratio to LoanToValue (LTV)

### PR DESCRIPTION
**Pull request by kamilovsky01**
Switch from SolvencyRatio to LoanToValue ratio.

In our case the LTV is equal to `Debt / Collateral`.
Collateral is equal to `total value - debt`.

### CODE REVIEW

File: contracts/SmartLoan.sol [hunk 1]
```diff
@@ -184,19 +188,19 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable {
...
       isSolvent() ? uint256(1) : uint256(0)
     ];
   }
 
 
   /**
     * Checks if the loan is solvent.
-    * It means that the ratio between total value and debt is above save level,
-    * which is parametrized by the minSolvencyRatio
+    * It means that the ratio between debt and collateral is below save level,
```
Comment (piotruda) [inline contracts/SmartLoan.sol:199]: `safe`

File: contracts/SmartLoan.sol [hunk 2]
```diff
@@ -20,12 +20,12 @@ import "redstone-flash-storage/lib/contracts/message-based/PriceAwareUpgradeable
...
 
   uint256 public constant PERCENTAGE_PRECISION = 1000;
-  uint256 private constant MAX_SOLVENCY_RATIO = 10000;
+  uint256 private constant MIN_LTV = 0;
 
   uint256 public constant LIQUIDATION_BONUS = 100;
   uint256 private constant LIQUIDATION_CAP = 200;
 
-  uint256 public constant minSolvencyRatio = 1200;
+  uint256 public constant maxLTVRatio = 5000;
```
Comment (piotruda) [inline contracts/SmartLoan.sol:28]: I would use an uppercase since we use it for other constants

File: contracts/SmartLoan.sol (hunk 1 above)
Comment (kamilovsky01) [reply contracts/SmartLoan.sol:199]: 👍

File: contracts/SmartLoan.sol [hunk 3]
```diff
@@ -170,12 +170,16 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable {
...
+    * The collateral is equal to total loan value takeaway debt.
+  **/
+  function getLoanToValueRatio() public view returns(uint256) {
     uint256 debt = getDebt();
     if (debt == 0) {
-      return MAX_SOLVENCY_RATIO;
+      return MIN_LTV;
     } else {
-      return getTotalValue() * PERCENTAGE_PRECISION / debt;
+      return debt * PERCENTAGE_PRECISION / (getTotalValue() - debt);
```
Comment (jakub-wojciechowski) [inline contracts/SmartLoan.sol:182]: We need to be careful here:
1) getTotalValue() == debt (division by 0)
2) getTotalValue() < debt (underflow)

Let's check that precondition before calculation and return MAX_LTV in that case

File: test/integration/smart-loan.test.ts [hunk 4]
```diff
@@ -109,21 +109,21 @@ describe('Smart loan', () => {
       await wrappedLoan.getTotalValue();
```
[the remark this answers is not in the document]
Comment (kamilovsky01) [reply test/integration/smart-loan.test.ts:109]: Done

File: contracts/mock/MockSmartLoan.sol [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    debt = _newDebt;
+  }
+
+
+  function setValue(uint256 _newValue) public {
+    value = _newValue;
+  }
+
+  /**
+   * Dummy implementation recording double deposits
```
Comment (jakub-wojciechowski) [inline contracts/mock/MockSmartLoan.sol:26]: Hmm, I don't think the comment fits in here ;)

File: test/integration/smart-loan.test.ts [hunk 6]
```diff
@@ -424,5 +424,47 @@ describe('Smart loan', () => {
...
+    });
+
+    it("should check debt greater than 0 and lesser than totalValue", async () => {
+      await loan.setValue(500);
+      await loan.setDebt(100);
+      expect(await loan.getLTV()).to.be.equal(250);
+      expect(await loan.isSolvent()).to.be.true;
+    });
+
+    it("should check debt equal to totalValue", async () => {
```
Comment (jakub-wojciechowski) [inline test/integration/smart-loan.test.ts:453]: Could we also 3 additional cases when:
1) Ltv is 4999
2) LTV is 5000 (calculated correctly)
3) LTV is 5001
?

**Pull request merged**